app/schema_validation/__init__.py

```diff
@@ -58,6 +59,32 @@ def validate_schema_date_with_hour(instance):
     return True
 
 
+@format_checker.checks('send_a_file_retention_period', raises=ValidationError)
+def validate_schema_retention_period(instance):
+    if instance is None:
+        return True
+
+    if isinstance(instance, str):
+        period = instance.strip().lower()
+        match = re.match(r'^(\d+) weeks?$', period)
+        if match and 1 <= int(match.group(1)) <= 78:
+            return True
+
+    raise ValidationError(
+        f"Unsupported value for retention_period: {instance}. Supported periods are from 1 to 78 weeks."
+    )
+
+
+@format_checker.checks('send_a_file_confirm_email', raises=ValidationError)
+def validate_schema_send_a_file_confirm_email(instance):
+    if instance is None or isinstance(instance, bool):
+        return True
+
+    raise ValidationError(
+        f"Unsupported value for confirm_email_before_download: {instance}. Use a boolean true or false value."
```

Comment:
do we need a custom error message for this? we don't have one for `is_csv` for example.

The default jsonschema error message is `'confirm_email_before_download potato is not of type boolean, null'`, which I think is probably fine?

If it's not fine, we might want to extend this validation/error message to apply to `is_csv` too.

Reply:
Just trying to go on what @karlchillmaid's come up with. I think this is opening up a wider issue around error messages for invalid inputs and this _probably_ isn't the time to have that wider discussion, though if it kicks that off I wouldn't be mad. I'm pretty sure with our current use of json schema we have a mix of content-designed error messaged and default JSON Schema ones.